pycloudlib/util.py

```diff
@@ -368,3 +370,19 @@ def get_query_param(uri: str, param: str) -> list:
     ['r134-fe06d70f']
     """
     return get_query_params(uri).get(param, [])
+
+
+def add_key_to_cloud_config(
+    public_key: str,
+    user_data: Optional[str] = None,
+):
+    """Add a public key to the cloud-config."""
+    if not user_data:
+        user_data_yaml = {"ssh_authorized_keys": [public_key]}
+    else:
+        user_data_yaml = yaml.safe_load(user_data)
```

Comment:
If user_data is something other than `#cloud-config` ... such as `## template: jinja` this function will inadvertently break that userdata by truncating the jinja header line and replacing it with just `#cloud-config`.   I'm not sure how much we care as it's likely we'll be running this initially manually/locally and on a subset of tests.... But, maybe it's worth checking the header line and raise ValueError if user_data isn't `#cloud-config` or at least `log.warning`?